### PR TITLE
[proxygen]: fix dependency

### PR DESCRIPTION
**Pull request by jiayuehua**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #...

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  <linux>, <Yes/No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  <Yes>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/proxygen/fix-zstd-mvfst-zlib-dependency.patch [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+diff --git a/cmake/proxygen-config.cmake.in b/cmake/proxygen-config.cmake.in
+index c5be55c..b803092 100644
+--- a/cmake/proxygen-config.cmake.in
++++ b/cmake/proxygen-config.cmake.in
+@@ -21,7 +21,7 @@ find_dependency(fmt)
+ find_dependency(folly)
+ find_dependency(wangle)
+ find_dependency(Fizz)
+-find_dependency(mvfst)
++#find_dependency(mvfst)
```
Comment (JackBoosY) [inline ports/proxygen/fix-zstd-mvfst-zlib-dependency.patch:23]: Why removed this dependency?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/proxygen/fix-zstd-mvfst-zlib-dependency.patch (hunk 1 above)
Comment (jiayuehua) [reply ports/proxygen/fix-zstd-mvfst-zlib-dependency.patch:23]: this dependency is not used.

Comment (JackBoosY) [reply ports/proxygen/fix-zstd-mvfst-zlib-dependency.patch:23]: Can you please confirm this with the upstream first?

Thanks.

Comment (jiayuehua) [reply ports/proxygen/fix-zstd-mvfst-zlib-dependency.patch:23]: ok https://github.com/facebook/proxygen/issues/396

**Review by ras0219-msft — APPROVED** [review]
This change 99% LGTM, however `zstd::libzstd_static` will only be defined if zstd was built statically.

Therefore, we either need to add logic to `proxygen/lib/CMakeLists.txt` to determine which target should be linked (`zstd::libzstd_static` vs `zstd::libzstd_shared`) or we should wait for a resolution to https://github.com/microsoft/vcpkg/pull/22822 which will hopefully define both targets always.

**Comment by jiayuehua** [comment]
I also think waiting for that patch is better. not merging now is right.

**Comment by JackBoosY** [comment]
Since https://github.com/microsoft/vcpkg/pull/22822 was merged, can we continue this PR?

**Comment by jiayuehua** [comment]
I will have a look soon. My daily job is a bit busy at every end of the seasons.

**Comment by jiayuehua** [comment]
hi, @JackBoosY . I have updated my pr.

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**